src/main/java/nova/core/component/misc/BlockMaterial.java

```diff
@@ -0,0 +1,124 @@
+package nova.core.component.misc;
+
+import nova.core.component.Component;
+import nova.core.sound.Sound;
+
+/**
+ * Created by winsock on 7/2/15.
+ */
+public class BlockMaterial extends Component {
+	protected boolean canBurn = false;
+	protected boolean replaceable = false;
+	protected boolean isTranslucent = false;
+	protected boolean requiresNoTool = true;
+	protected boolean blocksMovement = true;
+	protected boolean blocksLight = true;
+	protected boolean isSolid = true;
+	protected boolean isLiquid = false;
+	protected String blockSoundResource = "stone";
+
+	public boolean isLiquid() {
+		return isLiquid;
+	}
+
+	public boolean isSolid() {
+		return isSolid;
+	}
+
+	public boolean blocksLight() {
+		return blocksLight;
+	}
+
+	public boolean blocksMovement() {
+		return blocksMovement;
+	}
+
+	public boolean canBurn() {
+		return canBurn;
+	}
+
+	public BlockMaterial setCanBurn(boolean canBurn) {
+		this.canBurn = canBurn;
+		return this;
+	}
+
+	public boolean isReplaceable() {
+		return replaceable;
+	}
+
+	public BlockMaterial setReplaceable(boolean replaceable) {
+		this.replaceable = replaceable;
+		return this;
+	}
+
+	public boolean isTranslucent() {
+		return isTranslucent;
+	}
+
+	public BlockMaterial setIsTranslucent(boolean isTranslucent) {
+		this.isTranslucent = isTranslucent;
+		return this;
+	}
+
+	public boolean requiresNoTool() {
+		return requiresNoTool;
+	}
+
+	public BlockMaterial setRequiresNoTool(boolean requiresNoTool) {
+		this.requiresNoTool = requiresNoTool;
+		return this;
+	}
+
+	public void setBlocksMovement(boolean blocksMovement) {
+		this.blocksMovement = blocksMovement;
+	}
+
+	public void setBlocksLight(boolean blocksLight) {
+		this.blocksLight = blocksLight;
+	}
+
+	public void setSolid(boolean isSolid) {
+		this.isSolid = isSolid;
+	}
+
+	public void setLiquid(boolean isLiquid) {
+		this.isLiquid = isLiquid;
+	}
+
+	public String getBlockSound() {
+		return blockSoundResource;
+	}
+
+	public BlockSound getBreakSound() {
+		return new BlockSound("dig.");
+	}
+
+	public BlockSound getStepSound() {
+		return new BlockSound("step.");
+	}
+
+	public BlockSound getPlaceSound() {
+		return getBreakSound();
+	}
+
+	public void setBlockSound(String blockSoundResource) {
+		this.blockSoundResource = blockSoundResource;
+	}
+
+	public class BlockSound extends Sound {
+		private final String type;
+
+		public BlockSound(String type) {
+			super();
+			this.type = type;
+		}
+
+		@Override
+		public String getID() {
+			if (blockSoundResource.contains(":")) {
+				return blockSoundResource;
```

Comment:
This is a gray area IMO. The idea of sounds associated with blocks would be agnostic AFAIK. There might be a better solution here, however I didn't see it.